rpcserver.go

```diff
@@ -375,6 +379,48 @@ func internalRPCError(errStr, context string) *dcrjson.RPCError {
 	return dcrjson.NewRPCError(dcrjson.ErrRPCInternal.Code, errStr)
 }
 
+// rpcInvalidError is a convenience function to convert an invalid parameter
+// error to an RPC error with the appropriate code set.  It also logs the error
+// to the RPC server subsystem since internal errors really should not occur.
+// The context parameter is only used in the log message and may be empty if
+// it's not needed.
+func rpcInvalidError(errStr, context string) *dcrjson.RPCError {
+	logStr := errStr
```

Comment:
This code shouldn't be here in any of these functions except `rpcInternalError`.  If you read the comment (which, as an aside, is incorrect for this function as is anyways), it specifically says "since internal errors really should not occur."

A `dcrjson.ErrRPCInvalidParameter` is due to invalid arguments from the caller, so there is definitely no reason that should be logged as an error to the RPC server.